src/modules/services/postgres.nix

```diff
@@ -85,13 +85,30 @@ let
 
     if [[ ! -d "$PGDATA" ]]; then
       initdb ${lib.concatStringsSep " " cfg.initdbArgs}
-      ${setupInitialDatabases}
-
-      ${runInitialScript}
+      echo
+      echo "PostgreSQL init process complete; ready for start up."
+      echo
     fi
 
     # Setup config
     cp ${configFile} "$PGDATA/postgresql.conf"
+
+    if [[ ! -f "$PGDATA/.configured" ]]; then
+      touch "$PGDATA/.configured"
+      echo
+      echo "PostgreSQL appears unconfigured; going to run initialization scripts."
+      echo
+
+      pg_ctl -D "$PGDATA" -w start
```

Comment:
https://github.com/docker-library/postgres/blob/master/16/alpine3.18/docker-entrypoint.sh#L263-L265

Reply:
This, plus changing the default listen address to `127.0.0.1` (which should be safe), alongside changing the readiness probe to use TCP/IP instead of unix socket should fix the readiness probe problem.

Reply:
Alternatively the unix socket directory could be set to a temporary directory with `mktemp`, if we don't want to break backward compatibility. In my experience most coworkers expect an IP to be bound, given most of the people migrate from docker-compose postgres to devenv.

Reply:
I don't really want to break backwards compatibility. I made the change to listen on "no addresses", I didn't have to modify the default listener (although I generally agree with the expectation of listening on TCP) to make sure that nothing was broken.

I now create a temporary directory using `mktemp` to hold the temporary initialization server socket. It removes the directory when finished and a `trap` removes it when the script `EXIT`s.

I also removed the need for the "trailer file" to tell if the database has been configured. Instead, I use a variable and set it when you actually run `initdb`. I like this better because it would have tried to run initialization scripts on all existing databases. This could be good or bad, depending on the scripts.

Side bar: if you have several minutes worth of initialization scripts, `process-compose` will forcibly kill the `postgres` process and you'll be stuck with an incomplete database. It's not something I am going to tackle right now, but be aware that those conditions exist.